### PR TITLE
Switch logging to go-kit

### PR DESCRIPTION
**Pull request by Infra-Red**
Fixes https://github.com/prometheus/collectd_exporter/issues/83

/cc @brian-brazil

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -205,7 +210,7 @@ func (c collectdCollector) Collect(ch chan<- prometheus.Metric) {
 		for i := range vl.Values {
 			m, err := newMetric(vl, i)
 			if err != nil {
-				log.Errorf("newMetric: %v", err)
+				level.Error(c.logger).Log("msg", "newMetric", "err", err)
```
Comment (brian-brazil) [inline main.go]: This message could be better.

File: main.go [hunk 2]
```diff
@@ -245,13 +250,15 @@ func startCollectdServer(ctx context.Context, w api.Writer) {
 	if *collectdTypesDB != "" {
 		file, err := os.Open(*collectdTypesDB)
 		if err != nil {
-			log.Fatalf("Can't open types.db file %s", *collectdTypesDB)
+			level.Error(logger).Log("msg", "Can't open types.db file", "types", *collectdTypesDB)
```
Comment (brian-brazil) [inline main.go]: Include the err

File: main.go [hunk 3]
```diff
@@ -245,13 +250,15 @@ func startCollectdServer(ctx context.Context, w api.Writer) {
...
+			os.Exit(1)
 		}
 		defer file.Close()
 
 		typesDB, err := api.NewTypesDB(file)
 		if err != nil {
-			log.Fatalf("Error in parsing types.db file %s", *collectdTypesDB)
+			level.Error(logger).Log("msg", "Error in parsing types.db file", "types", *collectdTypesDB)
```
Comment (brian-brazil) [inline main.go]: and here

File: main.go (hunk 1 above)
Comment (Infra-Red) [reply main.go]: Fixed

File: main.go (hunk 2 above)
Comment (Infra-Red) [reply main.go]: Fixed

File: main.go (hunk 3 above)
Comment (Infra-Red) [reply main.go]: Fixed

**Comment by Infra-Red** [comment]
Hi @brian-brazil, I fixed the comments. Please let me know if anything else is required.

**Comment by brian-brazil** [comment]
That looks fine to me, @juliusv should give it the final review.

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**